### ISSUE TITLE
[sign_in_with_apple] web support

### ISSUE THREAD

**Issue by tapkain** [aboutyou/dart_packages#95]
Thanks for the package! Is it planned to have web support soon for it?

**Comment by HenriBeck** [comment]
@tapkain Probably not soon. Apple already offers a solution for the web stack, see #90 for a more detailed explanation.

**Comment by tapkain** [comment]
Thanks for the reference, will do that! Offer - would be great to include that info in the readme web section, so that future users won't bloat here the github issues.

**Comment by bdlukaa** [comment]
Hello. Would you like to reconsider adding web support to sign_in_with_apple? Flutter web is progressing and it'd be nice to have the option to [sign in with apple](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js) in it.

[Rest api support](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api) would also be nice :)

**Comment by christophemacabiau** [comment]
Thanks for your work on this package.
I agree with others, tt would be so nice to have web support for this package!

**Comment by truthmast** [comment]
I've recently been working on a interop library for Sign in with Apple JS: https://pub.dev/packages/sign_in_with_apple_js
Currently using it in conjunction with this package to provide Apple Sign In on all platforms.

Would a pull-request implementing web support using this package to implement web support be agreeable? Some extra steps I would include:
- Dynamically adding this script tag
```html
<script type="text/javascript" src="https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js"></script>
```
- Change plugin to enable web platform in different spots where it checks for iOS or Android
- Calling Sign in with Apple JS `init` and `signIn` from `getAppleIDCredential`

If this sounds good, I'll take whip up a PR.

**Comment by m-j-g** [comment]
> I've recently been working on a interop library for Sign in with Apple JS: https://pub.dev/packages/sign_in_with_apple_js
> Currently using it in conjunction with this package to provide Apple Sign In on all platforms.
> 
> Would a pull-request implementing web support using this interop package be agreeable? Some extra steps I would include:
> 
> * Creating a new package -- sign_in_with_apple_web -- with the interop package as a dependency
> * Copy the interface from `sign_in_with_apple`
> * Dynamically adding this script tag
> 
> ```
> <script type="text/javascript" src="https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js"></script>
> ```
> 
> * Calling Sign in with Apple JS `init` and `signIn` from `getAppleIDCredential`
> * Change this package to use `sign_in_with_apple_web` on web platforms
> 
> If this sounds good, I'll whip up a PR.
> 
> (( couple edits about implementation ))

Hey mhrst,
I'm trying to implement this but am a little bit naive about how to handle the communication here.  I already have sign_in_with_apple done for iOS and Android.  So, I have a server redirect url for Android that posts an intent back to the app.  I'm trying to understand what that should be for Web?  I have the simple pieces of code added in there, and so currently on Web I can launch the apple sign in window, sign in and then it gives them message about "Do you want to continue to use your Apple ID for this app" and when I click continue nothing happens.  Which is expected, bc I'm not sure how to send information from the server side redirect to the web app the same way that I do with Android.  Any advice or tips would be awesome!

**Comment by tp** [comment]
@mhrst et all: Revisiting this issue, I see how it could and should be much easier to get Sign in with Apple into web project.

If you're still up for it, let's work together and add it to this project so it'll work out of the box :)

**Comment by mhrst** [comment]
Thanks for your patience while I find the time to get started on this. Finally diving in!

I'll use this thread for any issues I run into, and to share progress. Looking to get it all completed by the end of this month. 👍🏼

**Comment by minusminuszero** [comment]
Hello! Any update on this yet?

**Comment by mhrst** [comment]
Delayed longer than I estimated -- thanks for everyone's patience 🙇🏽. Now moving along on this:

https://github.com/mhrst/dart_packages/tree/web

First commit refactors the plug-in as a [federated plugin](https://medium.com/flutter/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6). This could make other PRs less easily automatically merged as it moves some code around.

**Comment by tp** [comment]
@mhrst Have you tackled this further beyond the federated plugin split? (Will check that out in detail and would merge your commit if that's still up to date.)

We'll certainly be implementing web support within the next week, so would be interesting so share any existing work or learnings beforehand. (I'll be starting out to keep it simple and work with "just Flutter" (with a redirect) and avoid the external JS as outlined in #90).

**Comment by tp** [comment]
Implementation is now underway in https://github.com/aboutyou/dart_packages/pull/263

**Comment by tp** [comment]
This has now been published as [version 3.3.0](https://pub.dev/packages/sign_in_with_apple/versions/3.3.0). Please check it out and reach out in case of any trouble updating your app :)
